cmd/launcher/platform/pre06/exit.go

```diff
@@ -0,0 +1,22 @@
+package pre06
+
+import (
+	"github.com/buildpacks/lifecycle/cmd"
+	"github.com/buildpacks/lifecycle/cmd/launcher/platform/common"
+)
+
+var exitCodes = map[common.LifecycleExitError]int{
+	// launch phase errors: 700-799
+	common.LaunchError: 702, // LaunchError indicates generic launch error
+}
+
+func (p *pre06Platform) CodeFor(errType common.LifecycleExitError) int {
+	return CodeFor(errType)
```

Comment:
Why do we need to call this function and can't we just move lines 18-21 here?

Reply:
I wanted to keep the same pattern that we're using here: https://github.com/buildpacks/lifecycle/blob/819a998c4ab12ff5ebca07cb95f23c6ee6f35ac3/cmd/lifecycle/platform/v06/analyze.go#L9 where a later platform could just call the function directly instead of relying on a "previous platform".

Reply:
Since it seems we're all in agreement that getting rid of "previous platform" is a good step, I'll mark this as resolved for now.

Reply:
Following your last changes, the code in `analyze.go` was changed. Do you think it's worth changing the `CodeFor` code as well?

Reply:
Sorry, I saw your comment only after posting mine. What do you think?

Reply:
I wouldn't move `CodeFor` into `analyzer.go` because it doesn't pertain to the analyzer... it feels fine to leave it here, unless you have a better place for it.

Reply:
I didn't think about moving it to `analyzer.go`. 
Since `CodeFor` is exactly the same for all versions, and the only thing that changes is the `exitCodes` map, maybe we can put the function in a common place and pass the map to it.

Reply:
I'm not sure I follow :/ Could you give an example of how you would like the code to read?

Reply:
In `cmd/launcher/platform/common/exit.go`:

```
func CodeFor(errType common.LifecycleExitError, exitCodes map[common.LifecycleExitError]int) int {
	if code, ok := exitCodes[errType]; ok {
		return code
	}
	return cmd.CodeFailed
}
```

Then in `cmd/launcher/platform/[legacy&v06&v07]/exit.go`:
```
func (p *XXXPlatform) CodeFor(errType common.LifecycleExitError) int {
	return common.CodeFor(errType, exitCodes)
}
```

I think that in this way you're not copying the code for every version and it's also not weird to have the `CodeFor` twice in some of the files.

Reply:
I realized that since every platform (beside the legacy one) is using the same codes, we could just move that function into `common` for now. Please see e1c3bd8.

Reply:
On thinking about this more, what you proposed makes the most sense I think. Please see b6d9f87 - there is more duplication this way, but the code is more readable.